core/src/main/scala/cats/data/Chain.scala

```diff
@@ -358,6 +359,24 @@ sealed abstract class Chain[+A] {
       iterX.hasNext == iterY.hasNext
     }
 
+  /**
+   * Remove duplicates. Duplicates are checked using `Order[_]` instance.
+   */
+  def distinct[AA >: A](implicit O: Order[AA]): Chain[AA] = {
+    implicit val ord = O.toOrdering
+
+    var alreadyIn = TreeSet.empty[AA]
+
+    foldLeft(Chain.empty[AA]) { (elementsSoFar, b) =>
```

Comment:
Just a question of curiosity. Why are you using TreeSet instead of pure scala's Set?

Reply:
This was done for symmetry with `NonEmptyList#distinct` but it's a good question. `Set` uses a `HashSet` with sets bigger than 4 elements.  `HashSet#contains` has O(1) and `TreeSet#contains` has O(log(n)).

Changing `Set` would also remove the artificial dependency on `Order`. Was there a reason for this dependency when `NonEmptyList#distinct` was introduced?

Reply:
I wrote up a functional set implementation that should speed up `distinct` in the future and let it depend on `Eq` instead of `Order`. Closed issue here, was suggested to put up a PR on cats-collection instead. https://github.com/typelevel/cats/issues/2593

Reply:
Actually, it seems a structure like this already exists in cats-collections called ISet. I'd be curious to see the performance difference between TreeSet and ISet for this use case.